pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/elastic/export/tabular/TabularParticipantParser.java

```diff
@@ -0,0 +1,329 @@
+package org.broadinstitute.dsm.model.elastic.export.tabular;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.JsonNode;
+import org.broadinstitute.dsm.db.DDPInstance;
+import org.broadinstitute.dsm.export.WorkflowAndFamilyIdExporter;
+import org.broadinstitute.dsm.model.Filter;
+import org.broadinstitute.dsm.model.ParticipantColumn;
+import org.broadinstitute.dsm.model.elastic.export.tabular.renderer.TextValueProvider;
+import org.broadinstitute.dsm.model.elastic.export.tabular.renderer.ValueProviderFactory;
+import org.broadinstitute.dsm.model.participant.ParticipantWrapperDto;
+import org.broadinstitute.dsm.statics.ESObjectConstants;
+import org.broadinstitute.dsm.util.ElasticSearchUtil;
+import org.broadinstitute.dsm.util.proxy.jackson.ObjectMapperSingleton;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Parses a list of ParticipantDtos from elasticSearch into a map of column names to string values.
+ * Uses a DDP instance to query the question schema to determine which questions are multi-select,
+ * and therefore may be spit into multiple columns
+ * Usage:
+ * p = new ParticipantDtoParser([[list of columns from UI]], DDPInstance)
+ * List configs = p.generateExportConfigs();
+ * List participantMaps = p.parse(listOfParticipantsFromES);
+ */
+public class TabularParticipantParser {
+    private static final Logger logger = LoggerFactory.getLogger(TabularParticipantParser.class);
+    private static final String COLUMN_SELECTED = "1";
+    private static final String COLUMN_UNSELECTED = "0";
+    private static final String DATA_AS_MAP = "dataAsMap";
+    private final List<Filter> filters;
+    private final DDPInstance ddpInstance;
```

Comment:
`DDPInstance` is old class I would say, slowly we try to switch to `Dao/Dto` pairs, there is a class called `DDPInstanceDto` I would use this one if it is possible.

Reply:
agree that might make sense.  For now I'm just mirroring the structure of the other controllers that use this pattern.